platform/platform-filter-clientinfo/src/main/java/org/codice/ddf/platform/filter/clientinfo/ClientInfoInjector.java

```diff
@@ -0,0 +1,142 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.filter.clientinfo;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * This CXF filter extracts client-specific information and places it in a {@link ThreadLocal}
+ * instance so it can be forwarded to the {@code CatalogFramework} as an {@link AccessPlugin}.
+ */
+public class ClientInfoInjector implements Filter, AccessPlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientInfoInjector.class);
+
+    private static final String CLIENT_INFO_KEY = "client-info";
+
+    private final ThreadLocal<ClientInfoContainer> clientInfo;
+
+    public ClientInfoInjector() {
+        this.clientInfo = ThreadLocal.withInitial(new Supplier<ClientInfoContainer>() {
+            @Override
+            public ClientInfoContainer get() {
+                return null;
+            }
+        });
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void destroy() {}
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+            FilterChain filterChain) throws IOException, ServletException {
+        ClientInfoContainer infoContainer = new ClientInfoContainer(servletRequest);
+        clientInfo.set(infoContainer);
+        filterChain.doFilter(servletRequest, servletResponse);
+    }
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    //region - PASSTHROUGH METHODS WITHOUT PROCESSING -
```

Comment:
You should inject the client info into all requests.  Client info is useful for all request types.